core/src/main/java/com/linecorp/armeria/common/logging/ResponseLogLevelMapper.java

```diff
@@ -0,0 +1,92 @@
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.HttpStatusClass;
+import com.linecorp.armeria.common.annotation.Nullable;
+
+/**
+ * A {@link Function} that determines the {@link LogLevel} of an {@link HttpResponse} from a given
+ * {@link RequestLog}.
+ *
+ * @see LoggingDecoratorBuilder#responseLogLevelMapper(ResponseLogLevelMapper)
+ */
+@FunctionalInterface
+public interface ResponseLogLevelMapper extends Function<RequestLog, LogLevel> {
+
+    /**
+     * Creates a new {@link ResponseLogLevelMapper} which returns {@link LogLevel#DEBUG} when logging
+     * successful responses (e.g., no unhandled exception) and {@link LogLevel#WARN} if failure.
+     */
+    static ResponseLogLevelMapper of() {
+        return log -> log.responseCause() == null ? LogLevel.DEBUG : LogLevel.WARN;
+    }
+
+    /**
+     * Creates a new {@link ResponseLogLevelMapper} which returns the specified {@link LogLevel} if the given
+     * {@link RequestLog}'s status is equal to the specified {@link HttpStatus}.
+     */
+    static ResponseLogLevelMapper of(HttpStatus status, LogLevel logLevel) {
+        requireNonNull(status, "status");
+        requireNonNull(logLevel, "logLevel");
+        return log -> log.responseStatus() == status ? logLevel : null;
+    }
+
+    /**
+     * Creates a new {@link ResponseLogLevelMapper} which returns the specified {@link LogLevel} if the given
+     * {@link RequestLog}'s status belongs to the specified {@link HttpStatusClass}.
+     */
+    static ResponseLogLevelMapper of(HttpStatusClass statusClass, LogLevel logLevel) {
+        requireNonNull(statusClass, "statusClass");
+        requireNonNull(logLevel, "logLevel");
+        return log -> log.responseStatus().codeClass() == statusClass ? logLevel : null;
+    }
+
+    /**
+     * Returns a {@link LogLevel} for the given {@link RequestLog}. The {@code null} lets the next handler
+     * specified with {@link #orElse(ResponseLogLevelMapper)} map the {@link RequestLog}.
+     */
+    @Nullable
+    @Override
+    LogLevel apply(RequestLog log);
+
+    /**
+     * Returns a composed {@link ResponseLogLevelMapper} which represents a logical OR of this
+     * {@link ResponseLogLevelMapper} and the given {@code other}.
+     * If this {@link ResponseLogLevelMapper#apply(RequestLog)} returns {@code null}, then the other
+     * {@link ResponseLogLevelMapper} will be applied.
+     */
+    default ResponseLogLevelMapper orElse(ResponseLogLevelMapper other) {
+        requireNonNull(other, "other");
+        if (this == other) {
+            return this;
+        }
+        return log -> {
+            final LogLevel logLevel = ResponseLogLevelMapper.this.apply(log);
```

Comment:
Remove unnecessary qualifier?
```suggestion
            final LogLevel logLevel = apply(log);
```